### PR TITLE
feat: add deployment token name

### PR DESCRIPTION
**Pull request by m8vago**
- Restrict deploy tokens to be only available for rolling versions
- Add deploy token name
- Refactor notification handling
- Show deploy token name in audit log

### CODE REVIEW

File: web/crux-ui/locales/en/audit.json [hunk 1]
```diff
@@ -1,3 +1,5 @@
 {
+  "token": "Token",
+  "deploymentTokenInitials": "dt",
```
Comment (polaroi8d) [inline web/crux-ui/locales/en/audit.json:3]: dt?

File: web/crux/src/app/audit.logger/audit.logger.service.ts [hunk 2]
```diff
@@ -1,28 +1,72 @@
...
+      },
+      select: {
+        id: true,
+        deployment: {
+          select: {
+            version: {
+              select: {
+                project: {
+                  select: {
+                    teamId: true, // ♫♪♫ rockin' around, the christmas tree ♪♫♫
```
Comment (polaroi8d) [inline web/crux/src/app/audit.logger/audit.logger.service.ts:55]: What? 🌴 🎵

File: web/crux/src/app/audit/audit.mapper.ts [hunk 3]
```diff
@@ -11,6 +14,44 @@ export default class AuditMapper {
...
+  }
+
+  toDetailsDto(it: AuditLogWithDeploymentToken, identities: Map<string, Identity>): AuditLogDto {
+    const base: Omit<AuditLogDto, 'user' | 'name'> = {
+      actorType: this.actorTypeToDto(it.actorType),
+      context: it.context,
+      createdAt: it.createdAt,
+      event: it.event,
+      method: it.method,
+      // email: .email as string,
```
Comment (polaroi8d) [inline web/crux/src/app/audit/audit.mapper.ts:34]: Why is this commented out?

File: web/crux/src/app/audit/audit.mapper.ts [hunk 4]
```diff
@@ -11,6 +14,44 @@ export default class AuditMapper {
...
+      data: it.data as object,
+    }
+
+    if (it.actorType === 'user') {
```
Comment (polaroi8d) [inline web/crux/src/app/audit/audit.mapper.ts:38]: using `AUDIT_LOG_ACTOR_TYPE_VALUES`  enum values is better, no?

File: web/crux/prisma/schema.prisma [hunk 5]
```diff
@@ -371,17 +373,25 @@ enum AuditLogRequestMethodEnum {
...
+  user
+  deploymentToken
+}
 
-  team Team @relation(fields: [teamId], references: [id], onDelete: Cascade)
+model AuditLog {
+  id                String                     @id @default(uuid()) @db.Uuid
+  createdAt         DateTime                   @default(now()) @db.Timestamptz(6)
+  actorType         AuditLogActorTypeEnum
+  userId            String?                    @db.Uuid
```
Comment (polaroi8d) [inline web/crux/prisma/schema.prisma:385]: Using only one column to store actorId and use `actorId` as a name.

File: web/crux/src/app/audit/audit.mapper.ts (hunk 3 above)
Comment (m8vago) [reply web/crux/src/app/audit/audit.mapper.ts:34]: nice catch, thanks!

File: web/crux-ui/locales/en/audit.json (hunk 1 above)
Comment (m8vago) [reply web/crux-ui/locales/en/audit.json:3]: fixed with an avatar as we discussed in person

File: web/crux/prisma/schema.prisma (hunk 5 above)
Comment (m8vago) [reply web/crux/prisma/schema.prisma:385]: It is necessary because of the foreign key. :(

File: web/crux-ui/src/validations/deployment.ts [hunk 6]
```diff
@@ -18,5 +18,6 @@ export const createDeploymentSchema = updateDeploymentSchema.concat(
 export const copyDeploymentSchema = createDeploymentSchema
 
 export const createDeploymentTokenSchema = yup.object().shape({
+  name: yup.string().required(),
```
Comment (robot9706) [inline web/crux-ui/src/validations/deployment.ts:21]: We usually have a rule for names to be at least 3 characters, isn't that required here too?

**Review by robot9706 — APPROVED** [review]
Minor comment, otherwhise tested lgtm 🎄

**Pull request merged**